### ISSUE TITLE
how to merge tag input  in LanguagePairDataset in translation task

### ISSUE THREAD

**Issue by Ricardokevins** [pytorch/fairseq#3373]
### Before asking:   
1. search the issues.   
2. search the docs.    

#### What is your question?
I use translation task for summary generation. Original translation take "train.document.bin" and "train.summary.bin" as input. 
I have a tag for each document token, I want dataloader to read it and feed it  into model. How can I do？

for example(the data I have now):
document ：A B C D E 
tag : 1 2 3 4 5
summary: A B C

translation task in fairseq use MMapIndexedDataset(default indexed dataset) to load document and summary respectively And return LanguagePairDataset.  
Tag is not a kind of token but I have to load it and ensure alignment between annotate tag and document token at the same time.

#### Code
'''python
return LanguagePairDataset(
        src_dataset,
        src_dataset.sizes,
        src_dict,
        tgt_dataset,
        tgt_dataset_sizes,
        tgt_dict,
        left_pad_source=left_pad_source,
        left_pad_target=left_pad_target,
        align_dataset=align_dataset,
        eos=eos,
        num_buckets=num_buckets,
        shuffle=shuffle,
        pad_to_multiple=pad_to_multiple,
    )
'''
#### What have you tried?
I think maybe I need to generate another tag file and use fairseq-preprocess it together with ducument and summary which will create "document.bin"  "summary.bin"  and "tag.bin"
but how to ensure alignment between each tag and document token and insert the code to load the tag data.

#### What's your environment?

 - fairseq Version (master):
 - PyTorch Version (e.g., 1.0)
 - OS (Linux):
 - How you installed fairseq (source):
 - Build command you used (official command in README):
 - Python version: 3.7
